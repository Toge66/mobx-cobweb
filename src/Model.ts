/***************************************************
 * Created by nanyuantingfeng on 2020/6/2 12:53. *
 ***************************************************/
import { Model as _Model } from './datx'
import { withStorage } from './mixins/withStorage'
import { withNetActions } from './mixins/withNetActions'

/**** For DTS File (Don`t remove it) ****/
import { INetActionsMixinForModel } from './interfaces/INetActionsMixin'
import { IModelConstructor } from './datx'

export class Model extends withStorage(withNetActions(_Model)) {}
