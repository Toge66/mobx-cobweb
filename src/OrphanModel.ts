/***************************************************
 * Created by nanyuantingfeng on 2020/7/21 11:40. *
 ***************************************************/
import { Model } from './Model'
import { attribute } from './index'
import { ORPHAN_MODEL_ID_KEY, ORPHAN_MODEL_ID_VAL } from './helpers/consts'

export class OrphanModel extends Model {
  static enableAutoId = true

  static preprocess(data: any = {}) {
    data[ORPHAN_MODEL_ID_KEY] = ORPHAN_MODEL_ID_VAL
    return data
  }

  @attribute({ isIdentifier: true })
  public [ORPHAN_MODEL_ID_KEY]: string
}
