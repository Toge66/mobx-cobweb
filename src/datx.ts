/***************************************************
 * Created by nanyuantingfeng on 2020/7/28 09:28. *
 ***************************************************/
export * from '@issues-beta/datx'
