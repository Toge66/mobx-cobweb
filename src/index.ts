/***************************************************
 * Created by nanyuantingfeng on 2019/11/26 12:22. *
 ***************************************************/
export * from 'datx'
export * from './interfaces'
export * from './implements/helpers/model'

export type { ISkeletonCollection } from './interfaces/ISkeletonCollection'
export type { ISkeletonModel } from './interfaces/ISkeletonModel'
export type { ISkeletonView } from './interfaces/ISkeletonView'
export type { IRawResponse } from './interfaces/IRawResponse'
export type { IRequestOptions } from './interfaces/IRequestOptions'
export type { IResponseView } from './interfaces/IResponseView'

export { ResponseView } from './implements/ResponseView'
export { GenericModel } from './implements/GenericModel'
export { GenericView } from './implements/GenericView'
export { setNetworkAdapter } from './implements/NetworkUtils'
export { clearAllCache, clearCacheByType } from './implements/cache'

export { skeleton } from './mixin'
export * from './views'

export * from './implements/adapter/NetworkAdapter'
