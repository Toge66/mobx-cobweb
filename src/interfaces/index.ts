/***************************************************
 * Created by nanyuantingfeng on 2019/11/26 14:12. *
 ***************************************************/
export * from './types'
export * from './IRawResponse'
export * from './IRequestOptions'
export * from './IRawData'
export * from './IResponseHeaders'
export * from './INetworkAdapter'
