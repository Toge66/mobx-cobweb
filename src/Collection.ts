/***************************************************
 * Created by nanyuantingfeng on 2020/6/2 12:55. *
 ***************************************************/
import { Collection as _Collection, IModelConstructor, IType, PureModel, ICollectionConstructor } from 'datx'
import { withNetPatches } from './mixins/withNetPatches'
import { withStorage } from './mixins/withStorage'
import { INetPatchesCollectionMixin } from './interfaces/INetPatchesCollectionMixin'
import { IStorageCollectionMixin } from './interfaces/IStorageCollectionMixin'

const WithNetPatchesCollection: ICollectionConstructor<
  IStorageCollectionMixin<_Collection> & INetPatchesCollectionMixin<_Collection> & _Collection
> = withStorage(withNetPatches(_Collection))

export class Collection extends WithNetPatchesCollection {
  static register<T extends PureModel>(O: IModelConstructor<T>) {
    if (!this.types.find((Q) => Q.type === O.type)) {
      this.types.push(O)
    }
  }

  register<T extends PureModel>(O: IModelConstructor<T>) {
    ;((this.constructor as unknown) as Collection).register(O)
  }

  findOrphan<T extends PureModel>(model?: IType | IModelConstructor<T>): T {
    return this.findOne(model, '__$orphan_id__(0)')
  }
}
