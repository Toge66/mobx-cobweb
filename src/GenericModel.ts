/***************************************************
 * Created by nanyuantingfeng on 2019/11/26 12:22. *
 ***************************************************/
import { IModelConstructor, PureModel } from 'datx'
import { withNetActions } from './mixins/withNetActions'
import { INetActionsMixin } from './interfaces/INetActionsMixin'
const WithNetActionModel: IModelConstructor<INetActionsMixin<PureModel> & PureModel> = withNetActions(PureModel)

export class GenericModel extends WithNetActionModel {}
