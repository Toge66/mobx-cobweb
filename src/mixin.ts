/***************************************************
 * Created by nanyuantingfeng on 2019/11/26 12:22. *
 ***************************************************/
import {
  ICollectionConstructor,
  IModelConstructor,
  isCollection,
  isModel,
  isView,
  IViewConstructor,
  PureCollection,
  PureModel,
  View
} from 'datx'

import { decorateCollection } from './implements/decorateCollection'
import { decorateModel } from './implements/decorateModel'
import { decorateView } from './implements/decorateView'
import { ISkeletonCollection, ISkeletonModel, ISkeletonModelConstructor } from './interfaces'

export function skeleton<T extends PureModel>(
  BaseClass: IModelConstructor<T>
): ISkeletonModelConstructor<T & ISkeletonModel>

export function skeleton<T extends PureCollection>(
  BaseClass: ICollectionConstructor<T>
): ICollectionConstructor<T & ISkeletonCollection>

export function skeleton<T extends PureModel | PureCollection | View>(
  BaseClass: IModelConstructor<T> | ICollectionConstructor<T> | IViewConstructor<T>
) {
  if (isModel(BaseClass)) {
    return decorateModel(BaseClass as typeof PureModel)
  } else if (isCollection(BaseClass)) {
    return decorateCollection(BaseClass as typeof PureCollection)
  } else if (isView(BaseClass)) {
    return decorateView<T>(BaseClass as typeof View)
  }

  throw new Error('The instance needs to be a model, collection or a view')
}
