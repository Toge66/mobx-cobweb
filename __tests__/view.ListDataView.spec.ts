import { collection, useFixtureLimitByPOST, useFixturePUTById, useFixturesByGET } from './config'
import { ListDataView } from '../src'
import { autorun } from 'mobx'
import PropertySet from './models/PropertySet'
import Staff from './models/Staff'

describe('ListDataView', () => {
  let scope: any = null

  beforeEach(() => {
    scope = useFixturesByGET()
    useFixtureLimitByPOST(PropertySet.endpoint)
    useFixtureLimitByPOST(Staff.endpoint)
    collection.removeAll(PropertySet)
  })

  test('should be fetched data by `page`', async () => {
    const listDataView = new ListDataView<PropertySet>(PropertySet, collection)
    const jestFn = jest.fn(() => listDataView.isLoading)
    autorun(jestFn)

    await listDataView.search({
      selector: {
        limit: [0, 10]
      }
    })

    expect(listDataView.data).toBeInstanceOf(Array)
    expect(listDataView.data.length).toBe(10)

    expect(listDataView.data.length).toBe(10)
    expect(listDataView.data[1]).toBeInstanceOf(PropertySet)
    expect(listDataView.data).toMatchSnapshot()
    expect(listDataView.data[1].name).toBe('Solutions')

    await listDataView.next()
    expect(listDataView.data.length).toBe(10)
    expect(listDataView.data).toMatchSnapshot()
    expect(listDataView.data[1].name).toBe('Auto Loan Account')
    expect(jestFn).toBeCalledTimes(5)

    await listDataView.last()
    expect(listDataView.data.length).toBe(10)
    expect(listDataView.data).toMatchSnapshot()
    expect(listDataView.data[1].name).toBe('Supervisor')
    expect(listDataView.data[8].name).toBe('withdrawal')
    expect(jestFn).toBeCalledTimes(7)

    await listDataView.prev()
    expect(listDataView.data.length).toBe(10)
    expect(listDataView.data).toMatchSnapshot()
    expect(listDataView.data[1].name).toBe('best-of-breed')
    expect(listDataView.data[8].name).toBe('Berkshire')
    expect(listDataView.data[9].name).toBe('calculating')
    expect(jestFn).toBeCalledTimes(9)

    await listDataView.first()
    expect(listDataView.data.length).toBe(10)
    expect(listDataView.data).toMatchSnapshot()
    expect(listDataView.data[0].name).toBe('Accounts')
    expect(listDataView.data[1].name).toBe('Solutions')
    expect(jestFn).toBeCalledTimes(11)
  })

  test('should be fetched data by `infinite`', async () => {
    const listDataView = new ListDataView<PropertySet>(PropertySet, collection)
    const jestFn = jest.fn(() => listDataView.isLoading)
    autorun(jestFn)
    await listDataView.search({
      selector: {
        limit: [0, 10]
      }
    })

    await listDataView.infinite(10, 20)
    expect(listDataView.data.length).toBe(30)
    expect(listDataView.data[1]).toMatchSnapshot()
    expect(listDataView.data[1].label).toBe('withdrawal')
    expect(jestFn).toBeCalledTimes(5)

    await listDataView.infinite(20, 60)
    expect(listDataView.data.length).toBe(90)
    expect(listDataView.data[44].label).toBe('primary')
    expect(jestFn).toBeCalledTimes(7)
  })

  test('should be revert model', async () => {
    const listDataView = new ListDataView<Staff>(Staff, collection)
    const sdd = await listDataView.search({
      selector: {
        limit: [0, 10]
      }
    })

    const fn = jest.fn()
    const oo: string[] = []
    autorun(() => {
      fn()
      oo.push(listDataView.data[1].name)
    })
    const staff = collection.findOne(Staff, '6ae31b1e6dda')
    staff.name = '9999'
    useFixturePUTById(Staff.endpoint, 200, { errorCode: 404, errorMessage: 'x' })
    await staff.upsert().catch((e) => {
      expect(e.error).toEqual({ errorMessage: 'x', errorCode: 404 })
    })
    expect(listDataView.data[1].name).toBe('AGP')
    expect(oo).toEqual(['AGP', '9999', 'AGP'])
    expect(fn).toBeCalledTimes(3)
  })
})
