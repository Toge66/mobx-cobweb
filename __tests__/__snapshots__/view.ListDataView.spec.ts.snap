// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListDataView should be fetched data by \`infinite\` 1`] = `
Object {
  "__META__": Object {
    "MODEL_PERSISTED_FIELD": false,
    "collection": undefined,
    "fields": Object {
      "label": Object {
        "defaultValue": undefined,
        "referenceDef": false,
      },
      "name": Object {
        "defaultValue": undefined,
        "referenceDef": false,
      },
    },
    "id": -52,
    "type": "flow.PropertySet",
  },
  "label": "withdrawal",
  "name": "Solutions",
}
`;

exports[`ListDataView should be fetched data by \`page\` 1`] = `
Array [
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -1,
      "type": "flow.PropertySet",
    },
    "label": "Cambridgeshire",
    "name": "Accounts",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -2,
      "type": "flow.PropertySet",
    },
    "label": "withdrawal",
    "name": "Solutions",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -3,
      "type": "flow.PropertySet",
    },
    "label": "Practical Wooden Shirt",
    "name": "mint green",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -4,
      "type": "flow.PropertySet",
    },
    "label": "Falls",
    "name": "pixel",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -5,
      "type": "flow.PropertySet",
    },
    "label": "array",
    "name": "Isle of Man",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -6,
      "type": "flow.PropertySet",
    },
    "label": "Rubber",
    "name": "deploy",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -7,
      "type": "flow.PropertySet",
    },
    "label": "Synergistic",
    "name": "Home Loan Account",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -8,
      "type": "flow.PropertySet",
    },
    "label": "innovate",
    "name": "Checking Account",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -9,
      "type": "flow.PropertySet",
    },
    "label": "Tools",
    "name": "engage",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -10,
      "type": "flow.PropertySet",
    },
    "label": "applications",
    "name": "Dynamic",
  },
]
`;

exports[`ListDataView should be fetched data by \`page\` 2`] = `
Array [
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -11,
      "type": "flow.PropertySet",
    },
    "label": "Small",
    "name": "Soap",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -12,
      "type": "flow.PropertySet",
    },
    "label": "front-end",
    "name": "Auto Loan Account",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -13,
      "type": "flow.PropertySet",
    },
    "label": "Health",
    "name": "reinvent",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -14,
      "type": "flow.PropertySet",
    },
    "label": "firewall",
    "name": "Connecticut",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -15,
      "type": "flow.PropertySet",
    },
    "label": "North Dakota",
    "name": "secured line",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -16,
      "type": "flow.PropertySet",
    },
    "label": "quantify",
    "name": "Auto Loan Account",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -17,
      "type": "flow.PropertySet",
    },
    "label": "Kip",
    "name": "Ameliorated",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -18,
      "type": "flow.PropertySet",
    },
    "label": "Investor",
    "name": "Infrastructure",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -19,
      "type": "flow.PropertySet",
    },
    "label": "Investment Account",
    "name": "West Virginia",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -20,
      "type": "flow.PropertySet",
    },
    "label": "Berkshire",
    "name": "primary",
  },
]
`;

exports[`ListDataView should be fetched data by \`page\` 3`] = `
Array [
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -21,
      "type": "flow.PropertySet",
    },
    "label": "navigate",
    "name": "SQL",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -22,
      "type": "flow.PropertySet",
    },
    "label": "open-source",
    "name": "Supervisor",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -23,
      "type": "flow.PropertySet",
    },
    "label": "info-mediaries",
    "name": "Jordanian Dinar",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -24,
      "type": "flow.PropertySet",
    },
    "label": "Chief",
    "name": "Assurance",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -25,
      "type": "flow.PropertySet",
    },
    "label": "purple",
    "name": "Tasty Rubber Shoes",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -26,
      "type": "flow.PropertySet",
    },
    "label": "capacitor",
    "name": "Cheese",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -27,
      "type": "flow.PropertySet",
    },
    "label": "Metrics",
    "name": "Assurance",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -28,
      "type": "flow.PropertySet",
    },
    "label": "Investment Account",
    "name": "Lilangeni",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -29,
      "type": "flow.PropertySet",
    },
    "label": "Kids",
    "name": "withdrawal",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -30,
      "type": "flow.PropertySet",
    },
    "label": "payment",
    "name": "generate",
  },
]
`;

exports[`ListDataView should be fetched data by \`page\` 4`] = `
Array [
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -31,
      "type": "flow.PropertySet",
    },
    "label": "SDD",
    "name": "multi-state",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -32,
      "type": "flow.PropertySet",
    },
    "label": "Electronics",
    "name": "best-of-breed",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -33,
      "type": "flow.PropertySet",
    },
    "label": "Incredible Granite Gloves",
    "name": "interfaces",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -34,
      "type": "flow.PropertySet",
    },
    "label": "connecting",
    "name": "Assistant",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -35,
      "type": "flow.PropertySet",
    },
    "label": "microchip",
    "name": "Associate",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -36,
      "type": "flow.PropertySet",
    },
    "label": "Oregon",
    "name": "Savings Account",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -37,
      "type": "flow.PropertySet",
    },
    "label": "Computer",
    "name": "Reverse-engineered",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -38,
      "type": "flow.PropertySet",
    },
    "label": "Administrator",
    "name": "Metal",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -39,
      "type": "flow.PropertySet",
    },
    "label": "enhance",
    "name": "Berkshire",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -40,
      "type": "flow.PropertySet",
    },
    "label": "Awesome Metal Mouse",
    "name": "calculating",
  },
]
`;

exports[`ListDataView should be fetched data by \`page\` 5`] = `
Array [
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -41,
      "type": "flow.PropertySet",
    },
    "label": "Cambridgeshire",
    "name": "Accounts",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -42,
      "type": "flow.PropertySet",
    },
    "label": "withdrawal",
    "name": "Solutions",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -43,
      "type": "flow.PropertySet",
    },
    "label": "Practical Wooden Shirt",
    "name": "mint green",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -44,
      "type": "flow.PropertySet",
    },
    "label": "Falls",
    "name": "pixel",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -45,
      "type": "flow.PropertySet",
    },
    "label": "array",
    "name": "Isle of Man",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -46,
      "type": "flow.PropertySet",
    },
    "label": "Rubber",
    "name": "deploy",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -47,
      "type": "flow.PropertySet",
    },
    "label": "Synergistic",
    "name": "Home Loan Account",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -48,
      "type": "flow.PropertySet",
    },
    "label": "innovate",
    "name": "Checking Account",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -49,
      "type": "flow.PropertySet",
    },
    "label": "Tools",
    "name": "engage",
  },
  Object {
    "__META__": Object {
      "MODEL_PERSISTED_FIELD": false,
      "collection": undefined,
      "fields": Object {
        "label": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
        "name": Object {
          "defaultValue": undefined,
          "referenceDef": false,
        },
      },
      "id": -50,
      "type": "flow.PropertySet",
    },
    "label": "applications",
    "name": "Dynamic",
  },
]
`;
